### ISSUE TITLE
cpanel migrate - CGI folder and some scripts have been setup

### ISSUE THREAD

**Issue by shoulders** [virtualmin/virtualmin-gpl#940]
| SYSTEM INFORMATION ||
|----------|----------|
|  OS type and version |  Ubuntu Linux 22.04.5             |
|  Usermin version     |  2.102                            |
|  Virtualmin version  |  7.20.2 Pro                       |
|  Theme version       |  21.20.7                          |
|  Apache version      |  2.4.52                           |
|  Package updates     |  8 package updates are available  |

## the issue

I imported a cpanel account in to virtualmin and it has created the following folder and CGI scripts

![image](https://github.com/user-attachments/assets/e0a385bd-56df-49fc-b0cd-3c8118ea4bb4)

- I do not have CGI installed and this is not in my server template.
- The PHP version is currently PHP-FPM
- I did notice during the import procedure it said it was setting up some CGI stuff but I just ignored it because I did not know if this was normal

![image](https://github.com/user-attachments/assets/06f4bcdf-fd3d-42e8-9b2f-70846c62ecc5)

![image](https://github.com/user-attachments/assets/404ffb9b-affa-403d-96d0-a42de3bf22f5)

### proposed solution

Don't install CGI stuff when CGI is not selected.

**Comment by jcameron** [comment]
Hmm, that's odd that those CGI wrappers would be created when they aren't needed! 

To help debug this, it would be useful to see the output from the migration process. Also, if you create a new domain on your system, does it get those `php*.cgi` files?

**Comment by shoulders** [comment]
>it would be useful to see the output from the migration process.

Is this the text that appears on screen while migrating?

I will runs some tests, clarify and get you the information.

**Comment by jcameron** [comment]
Hmmm ... I'm not sure what could be going on here then. Maybe the default PHP mode wasn't set to FPM when a domain was imported, and thus those `.cgi` files were created?

**Comment by shoulders** [comment]
I don't have the php CGI mode available as I removed it. I think I removed  php-cgi module for each php version.

I did have them installed at one point, I think.

Only php-fpm is available.

It is almost like some of the behind the scenes handling of php-cgi is being triggered as if it were installed.

Parts of the template system might be being ignored for php when you import.

Are there hardcoded rules for handling CGI?

Is the import saying cpanel had CGI so I will now add CGI files?

Tricky one. I can't see anything obvious.

**Comment by jcameron** [comment]
No, we don't even get the PHP mode from the cPanel backup. Hard to say what's going on here unless it can be re-produced, as I don't see any obvious trigger for this in the code.

**Comment by iliajie** [comment]
> I did find one code path where the `.cgi` wrappers can be created unnecessarily. I'll fix this for the next release : [4e3efd2](https://github.com/virtualmin/virtualmin-gpl/commit/4e3efd2bfaff558d1d652b240f4a0e5dfd1fba93)

Even with this patch, if this function is called, like in FPM mode, it will still default to `&cgi_bin_dir($_[0])`, and create `~/cgi-bin` and stuff, right?

Why not just add after `$mode`:

```Perl
return if ($mode !~ /cgi/);
```

**Comment by jcameron** [comment]
But cgi-bin directories are different from PHP .cgi wrappers.

I will clean up those `if` statements though.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run. @shoulders, sorry for the noise, and thanks for the contribution. We definitely didn't mean any disrespect.

If it turns out the issue is already resolved for you, you're welcome to close it.
